src/station_gui.cpp

```diff
@@ -260,8 +260,8 @@ class CompanyStationsWindow : public Window
 		this->stations_per_cargo_type_no_rating = 0;
 
 		for (const Station *st : Station::Iterate()) {
-			if (st->owner == owner || (st->owner == OWNER_NONE && HasStationInUse(st->index, true, owner))) {
-				if (this->filter.facilities & st->facilities) { // only stations with selected facilities
+			if ((this->filter.facilities & st->facilities) != 0) { // only stations with selected facilities
```

Comment:
while at it, mind making this a bit more readable by using `continue` instead of this deep nesting?

Reply:
Will do as a separate PR I think, to avoid obscuring this simple change.